pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -726,12 +726,16 @@ func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings,
 
 	// DO not do if else here. since lb.GetSimple returns a enum value (not pointer).
 	switch lb.GetSimple() {
+	case networking.LoadBalancerSettings_ROUND_ROBIN:
```

Comment:
I get why you are doing this but it seems problematic that a user cannot set ROUND_ROBIN explicitly

Reply:
The reason being that RR is not generally safe and should never really be used. I don't think we should even allow it.

Reply:
I don't think removing options is a good thing. Changing the default is good but we don't need to remove options just because they are not the ideal in standard scenarios.

Reply:
I generally understand where you're coming from, but I suspect nobody chooses RR. They just either use the default (probably 99.9% of cases) or select it because they don't particularly know or care which algorithm is used. Why would anyone ever want RR?

Reply:
How would this break backward compatibility?

Reply:
Today: I set ROUND_ROBIN and get ROUND_ROBIN
Future: I set ROUND_ROBIN and do not get ROUND_ROBIN

I suppose it doesn't strictly break compat right now since its off by default but the plan is to make it default

Reply:
If the user needs RR, they will still be able to disable this option (assuming it's eventually enabled by default).

@louiscryan @smawson @costinm any thoughts?

Reply:
So when it is eventually enabled by default - do we allow ROUND_ROBIN? I think there are valid usecases for ROUND_ROBIN. In our environment we set LEAST_CONN by default but certain services (one usecase is  one service has to connect its peers in a round robin way) use this. So removing the option is not good.

Reply:
FWIW I have been preaching the `LEAST_CONN` gospel for years over in Envoy-land and still don't think it's a good idea to disallow WRR completely. Without the option to configure RR, there's no way to guarantee strict weighted host selection.

Just my 2 cents- I don't really have a horse in this race.